### PR TITLE
feat: add default policies

### PR DESCRIPTION
**Pull request by stebenz**

### CODE REVIEW

File: zitadel/v2/default_login_policy/funcs.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+	client, err := helper.GetAdminClient(clientinfo)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	current, err := client.GetLoginPolicy(ctx, &admin.GetLoginPolicyRequest{})
+	if err != nil {
+		return diag.FromErr(err)
+	}
```
Comment (eliobischof) [inline zitadel/v2/default_login_policy/funcs.go:38]: Why don't we compare to the TF state?

File: zitadel/v2/default_domain_policy/funcs.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/zitadel/terraform-provider-zitadel/zitadel/v2/helper"
+)
+
+func delete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	tflog.Info(ctx, "default domain policy cannot be deleted")
+	return nil
+}
+
+func update(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	tflog.Info(ctx, "started update")
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_domain_policy/funcs.go:20]: debug what? Do you mean with that it should be DEBUG-level not INFO-level log?

File: zitadel/v2/default_domain_policy/funcs.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	})
+	if err != nil {
+		return diag.Errorf("failed to update default domain policy: %v", err)
+	}
+	d.SetId(resp.GetDetails().GetResourceOwner())
+	return nil
+}
+
+func read(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	tflog.Info(ctx, "started read")
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_domain_policy/funcs.go:45]: debug what? Do you mean with that it should be DEBUG-level not INFO-level log?

File: zitadel/v2/default_domain_policy/funcs.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+	client, err := helper.GetAdminClient(clientinfo)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := client.GetDomainPolicy(ctx, &admin.GetDomainPolicyRequest{})
+	if err != nil {
+		d.SetId("")
+		return nil
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_domain_policy/funcs.go:60]: what just if 404?

File: zitadel/v2/default_label_policy/funcs.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+		FontColorDark:       d.Get(fontColorDarkVar).(string),
+		DisableWatermark:    d.Get(disableWatermarkVar).(bool),
+	})
+	if err != nil {
+		return diag.Errorf("failed to update default label policy: %v", err)
+	}
+	d.SetId(resp.Details.ResourceOwner)
+
+	active := d.Get(setActiveVar)
+	if active != nil {
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_label_policy/funcs.go:50]: you mean to use active=true as default? wouldn't do that as this flag is meant to give the possibility to set inactive label policies

File: zitadel/v2/default_domain_policy/funcs.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/v2/default_domain_policy/funcs.go:45]: exactly :-D

File: zitadel/v2/default_label_policy/resource.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+			primaryColorDarkVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "hex value for primary color dark theme",
+			},
+			backgroundColorDarkVar: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "hex value for background color dark theme",
+			},
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_label_policy/resource.go:45]: could be defined optional, with additional code in the update function

File: zitadel/v2/default_login_policy/funcs.go (hunk 1 above)
Comment (stebenz) [reply zitadel/v2/default_login_policy/funcs.go:38]: compare what to the TF state?

File: zitadel/v2/default_domain_policy/funcs.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/v2/default_domain_policy/funcs.go:60]: return nil, just if the resource does not exist. other api errors should be passed through

[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_domain_policy/funcs.go:60]: Hm yeah, that would be great, but the problem is how should I differentiate the errors?

File: zitadel/v2/default_label_policy/funcs.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/v2/default_label_policy/funcs.go:50]: i'm pretty sure that most terraform users, who declare the resources config, also would like to have it activated. What do you think? I just see a source for confusion in defaulting this to false.

File: zitadel/v2/default_login_policy/funcs.go (hunk 1 above)
Comment (eliobischof) [reply zitadel/v2/default_login_policy/funcs.go:38]: To the declared state. Comparision is made here: https://github.com/zitadel/terraform-provider-zitadel/pull/36/files#diff-c3f8681c64238c3b31799bf985066776ec5d64f242fc39cabe4bae2fb1bf97e9R71

File: zitadel/v2/default_domain_policy/funcs.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_domain_policy/funcs.go:45]: terraform logs are only visible if you set an environment variable TF_LOG to "info" or "debug", so IMO info is correct for me, debug would include some additional information about request etc.

File: zitadel/v2/default_label_policy/funcs.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_label_policy/funcs.go:50]: for me setting something like a design to active automatically would be incorrect, as maybe you want to apply it or import it, and still have to differentiate if it is set active as this is also an attribute for the resource

File: zitadel/v2/default_domain_policy/funcs.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/v2/default_domain_policy/funcs.go:60]: @livio-a doesn't the grpc api return proper status codes?

[the remark this answers is not in the document]
Comment (eliobischof) [reply zitadel/v2/default_domain_policy/funcs.go:60]: According to the docs, you should be able to differenciate between already exists, not found and other by the status https://docs.zitadel.com/docs/apis/statuscodes

File: zitadel/v2/default_label_policy/funcs.go [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+		activeBool := active.(bool)
+		if activeBool {
+			if _, err := client.ActivateLabelPolicy(ctx, &admin.ActivateLabelPolicyRequest{}); err != nil {
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/default_label_policy/funcs.go:64]: now has a check if changed, so shouldn't be activated if already active

**Review by eliobischof — APPROVED** [review]

**Pull request merged**